ecs-logs-collector.sh

```diff
@@ -428,52 +460,70 @@ get_docker_info()
     timeout 20 docker version > ${info_system}/docker/docker-version.txt 2>&1 || echo "Timed out, ignoring \"docker version output \" "
 
     ok
-
   else
-    echo "WARNING: The Docker daemon is not running." | tee ${info_system}/docker/docker-not-running.txt
+    warning "the Docker daemon is not running." | tee ${info_system}/docker/docker-not-running.txt
   fi
 }
 
-get_containers_info()
-{
-  try "inspect running Docker containers and gather Amazon ECS container agent data"
-  pgrep agent > /dev/null
+get_ecs_agent_info() {
+  try "collect Amazon ECS Container Agent state and config"
 
-  if [[ "$?" -eq 0 ]]; then
-    mkdir -p ${info_system}/docker
+  mkdir -p ${info_system}/ecs-agent
+  if [ -e /var/lib/ecs/data/ecs_agent_data.json ]; then
+    cat  /var/lib/ecs/data/ecs_agent_data.json | python -mjson.tool > ${info_system}/ecs-agent/ecs_agent_data.txt 2>&1
+  fi
 
-    for i in `docker ps -q`; do
-      docker inspect $i > $info_system/docker/container-$i.txt 2>&1
-      if grep --quiet "ECS_ENGINE_AUTH_DATA" $info_system/docker/container-$i.txt; then
-        sed -i 's/ECS_ENGINE_AUTH_DATA=.*/ECS_ENGINE_AUTH_DATA=/g' $info_system/docker/container-$i.txt
-      fi
-    done
+  if [ -e /etc/ecs/ecs.config ]; then
+    cp -f /etc/ecs/ecs.config ${info_system}/ecs-agent/ 2>&1
+    if grep --quiet "ECS_ENGINE_AUTH_DATA" ${info_system}/ecs-agent/ecs.config; then
+      sed -i 's/ECS_ENGINE_AUTH_DATA=.*/ECS_ENGINE_AUTH_DATA=/g' ${info_system}/ecs-agent/ecs.config
+    fi
+  fi
+  ok
 
+  try "collect Amazon ECS Container Agent engine data"
 
-    if [ -e /usr/bin/curl ]; then
+  pgrep agent > /dev/null
+  if [[ "$?" -eq 0 ]]; then
+    if command -v curl >/dev/null; then
```

Comment:
We've got this same test elsewhere. I wonder if we could perform it once, early in the program, exit if it fails (suggesting that the user install curl in that case), and continue if it passes.

Reply:
@zmrow pointed out that we might want to establish the presence of required and optional tools that we require to collect various data points. This would better handle the redundant checks and enable this script to gracefully fallback on to alternative checks if possible. See https://github.com/awslabs/ecs-logs-collector/issues/35